src/payment/checkout/payment-form/StripePaymentForm.test.jsx

```diff
@@ -0,0 +1,283 @@
+
+jest.mock('@edx/frontend-platform/analytics', () => ({
+  sendTrackEvent: jest.fn(),
+}));
+
+jest.useFakeTimers('modern');
+
+const validateRequiredFieldsMock = jest.spyOn(formValidators, 'validateRequiredFields');
+
+const mockStore = configureMockStore();
+
+configureI18n({
+  config: {
+    ENVIRONMENT: process.env.ENVIRONMENT,
+    LANGUAGE_PREFERENCE_COOKIE_NAME: process.env.LANGUAGE_PREFERENCE_COOKIE_NAME,
+  },
+  loggingService: {
+    logError: jest.fn(),
+    logInfo: jest.fn(),
+  },
+  messages: {
+    uk: {},
+    th: {},
+    ru: {},
+    'pt-br': {},
+    pl: {},
+    'ko-kr': {},
+    id: {},
+    he: {},
+    ca: {},
+    'zh-cn': {},
+    fr: {},
+    'es-419': {},
+    ar: {},
+  },
+});
+
+const authenticatedUser = Factory.build('userAccount');
+
+describe('<StripePaymentForm />', () => {
+  let store;
+  let mockStripe;
+  let mockElements;
+  let submitButton;
+  let state;
+  let mockElement;
+  let paymentElement;
+
+  beforeEach(() => {
+    store = createStore(createRootReducer(), {});
+    mockStripe = mocks.mockStripe();
+    mockElement = mocks.mockElement();
+    mockElements = mocks.mockElements();
+    mockStripe.elements.mockReturnValue(mockElements);
+    mockElements.create.mockReturnValue(mockElement);
+
+    state = {
+      payment: {
+        basket: {
+          loading: true,
+          loaded: false,
+          submitting: false,
+          redirect: false,
+          isBasketProcessing: false,
+          products: [{ sku: '00000' }],
+          enableStripePaymentProcessor: true,
+        },
+        clientSecret: { isClientSecretProcessing: false, clientSecretId: '' },
+      },
+      feedback: { byId: {}, orderedIds: [] },
+      form: {},
+    };
+
+    const wrapper = mount((
+      <IntlProvider locale="en">
+        <AppContext.Provider value={{ authenticatedUser }}>
+          <reactRedux.Provider store={mockStore(state)}>
+            <Elements stripe={mockStripe}>
+              <StripePaymentForm
+                handleSubmit={() => {}}
+                onSubmitPayment={() => {}}
+                onSubmitButtonClick={() => {}}
+              />
+            </Elements>
+          </reactRedux.Provider>
+        </AppContext.Provider>
+      </IntlProvider>
+    ));
+    paymentElement = wrapper.find(PaymentElement);
+    submitButton = wrapper.find(PlaceOrderButton);
+  });
+
+  afterEach(() => {
+    jest.restoreAllMocks();
+  });
+
+  describe('load stripe', () => {
+    it('creates the payment element', () => {
+      expect(mockElements.create).toHaveBeenCalledWith('payment', null);
+      expect(paymentElement.length > 0).toBe(true);
+    });
+  });
+
+  describe('getRequiredFields', () => {
+    it('returns expected required fields', () => {
+      const testFormValues = [
+        {
+          firstName: '',
+          lastName: '',
+          address: '',
+          city: '',
+          country: 'UK',
+          cardExpirationMonth: '',
```

Comment:
We don't have these fields as part of our custom form for Stripe, should they be removed? CyberSource does have the custom credit card fields.

Reply:
removed